src/sqlfluff/core/linter/linter.py

```diff
@@ -494,112 +495,142 @@ def lint_fix_parsed(
+                progress_bar_crawler = tqdm(
+                    rules_this_phase,
+                    desc="lint by rules",
+                    leave=False,
+                    disable=progress_bar_configuration.disable_progress_bar,
                 )
-                all_linting_errors += linting_errors
-
-                if fix and fixes:
-                    linter_logger.info(f"Applying Fixes [{crawler.code}]: {fixes}")
-                    # Do some sanity checks on the fixes before applying.
-                    anchor_info = BaseSegment.compute_anchor_edit_info(fixes)
-                    if any(
-                        not info.is_valid for info in anchor_info.values()
-                    ):  # pragma: no cover
-                        message = (
-                            f"Rule {crawler.code} returned conflicting fixes with the "
-                            f"same anchor. This is only supported for create_before+"
-                            f"create_after, so the fixes will not be applied. {fixes!r}"
-                        )
-                        cls._report_conflicting_fixes_same_anchor(message)
-                        for lint_result in linting_errors:
-                            lint_result.fixes = []
-                    elif fixes == last_fixes:  # pragma: no cover
-                        # If we generate the same fixes two times in a row,
-                        # that means we're in a loop, and we want to stop.
-                        # (Fixes should address issues, hence different
-                        # and/or fewer fixes next time.)
-                        cls._warn_unfixable(crawler.code)
-                    else:
-                        # This is the happy path. We have fixes, now we want to
-                        # apply them.
-                        last_fixes = fixes
-                        new_tree, _ = tree.apply_fixes(
-                            config.get("dialect_obj"), crawler.code, anchor_info
-                        )
-                        # Check for infinite loops
-                        if new_tree.raw not in previous_versions:
-                            # We've not seen this version of the file so far. Continue.
-                            tree = new_tree
-                            previous_versions.add(tree.raw)
-                            changed = True
-                            continue
-                        else:
-                            # Applying these fixes took us back to a state which we've
-                            # seen before. We're in a loop, so we want to stop.
-                            cls._warn_unfixable(crawler.code)
 
-            if loop == 0:
-                # Keep track of initial errors for reporting.
-                initial_linting_errors = all_linting_errors.copy()
+                for crawler in progress_bar_crawler:
+                    # Performance: After first loop pass, skip rules that don't
```

Comment:
@tunetheweb asked in Slack: I’m still confused why we don’t need these in the final phase (which is a post) for any new code added by other rules

Reply:
This is a good point. I think you could call this a design decision, and the linter has **always** behaved this way. When the user runs `fix`, they only see `initial_linting_errors`, not `all_linting_errors`.

For more context, (`crawl()`) computes two sets of errors:
* `initial_linting_errors`: This is what the function actually returns. It is assigned exactly once. Old behavior: Assigned on loop 1. New behavior: Assigned on loop 1 of the main phase (basically same behavior). Note that loop 1 of the main phase runs all the rules, including the post rules. I'll add PR comments to these places with additional comments.
* `all_linting_errors`:

Reply:
I went back and refactored how this works. The old code was unnecessarily tricky -- the `all_linting_errors` was really not used after the first pass, so why have it? 🤯 

Note that this PR doesn't change any tests except for those two oddball "runaway" tests, so if the tests pass, I think we can be fairly confident that changing the bookkeeping didn't break anything. (In fact, several of the little changes inside the linter loop that we've been discussing were added because some tests failed, i.e. caught some initial bugs. 🎉 🥳 )